src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/risk/converter/RiskScoreConverter.kt

```diff
@@ -0,0 +1,140 @@
+      historical = historicalScores.mapNotNull { convert(it) },
+    )
+  }
+
+  /**
+   * Convert a RiskScoreResponse to a PredictorScore object
+   */
+  fun convert(riskScoreResponse: RiskScoreResponse): PredictorScore? {
+    val scores = createScores(riskScoreResponse)
+    if (scores?.ogp == null
+      && scores?.ogrs == null
+      && scores?.ovp == null
+      && scores?.rsr == null
+      && scores?.ospc == null
+      && scores?.ospi == null
+    ) {
+      return null
+    } else {
+      return PredictorScore(
+        date = convertDateTimeStringToDateString(riskScoreResponse.completedDate),
+        scores = scores,
+      )
+    }
+  }
+
+  private fun convertDateTimeStringToDateString(dateTime: String?): String? {
+    return if (dateTime != null) {
+      LocalDateTime.parse(dateTime).toLocalDate().toString()
+    } else {
+      null
+    }
+  }
+
+  private fun createScores(riskScoreResponse: RiskScoreResponse): Scores? {
+    return Scores(
+      rsr = rsrLevelWithScore(riskScoreResponse),
+      ospc = buildLevelWithScore(
+        riskScoreResponse.sexualPredictorScore?.ospContactScoreLevel,
+        riskScoreResponse.sexualPredictorScore?.ospContactPercentageScore,
+        "OSP/C",
+      ),
+      ospi = buildLevelWithScore(
+        riskScoreResponse.sexualPredictorScore?.ospIndecentScoreLevel,
+        riskScoreResponse.sexualPredictorScore?.ospIndecentPercentageScore,
+        "OSP/I",
+      ),
+      ogrs = buildTwoYearScore(
+        riskScoreResponse.groupReconvictionScore?.scoreLevel,
+        riskScoreResponse.groupReconvictionScore?.oneYear,
+        riskScoreResponse.groupReconvictionScore?.twoYears,
+        "OGRS",
+      ),
+      ogp = buildTwoYearScore(
+        riskScoreResponse.generalPredictorScore?.ogpRisk,
+        riskScoreResponse.generalPredictorScore?.ogp1Year,
+        riskScoreResponse.generalPredictorScore?.ogp2Year,
+        "OGP",
+      ),
+      ovp = buildTwoYearScore(
+        riskScoreResponse.violencePredictorScore?.ovpRisk,
+        riskScoreResponse.violencePredictorScore?.oneYear,
+        riskScoreResponse.violencePredictorScore?.twoYears,
+        "OVP",
+      ),
+    )
+  }
+
+  private fun rsrLevelWithScore(riskScoreResponse: RiskScoreResponse?): LevelWithScore? {
+    val rsr = riskScoreResponse?.riskOfSeriousRecidivismScore
+    val rsrScore = riskScoreResponse?.riskOfSeriousRecidivismScore
+    return if (rsrScore?.scoreLevel == null && rsrScore?.percentageScore == null) {
+      null
+    } else {
+      LevelWithScore(
+        level = rsr?.scoreLevel,
+        score = rsr?.percentageScore,
+        type = "RSR",
+      )
+    }
+  }
+
+  private fun buildLevelWithScore(
+    level: String?, percentageScore: String?, type: String?,
+  ): LevelWithScore? {
+    val scoreIsNull = level == null && percentageScore == null
+    val notApplicableWithZeroPercentScorePresent =
+      level.equals(SCORE_NOT_APPLICABLE, ignoreCase = true) && percentageScore == "0"
```

Comment:
Do these percentageScores come in from ARNs as strings? Would it be beneficial to cast them to numerical types once we have them?

Reply:
I wondered about this, but thought I'd discuss with you at some point to do in the future. Didn't want to make this a larger PR than it is already. I guess it's also debatable whether there is any benefit in converting it when all we're going to do is pass it through to the UI - either way, for now I don't think it does any harm leaving it as it was, so won't change it here.